portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,21 +31,44 @@
 import javax.servlet.http.HttpServletResponse;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
-
-    ArrayList<String> messages = new ArrayList<>(Arrays.asList("Hello", "Hola", "Bonjour"));
-
     @Override
     public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        ArrayList<String> messages = new ArrayList<>();
+        for(Entity entity : results.asIterable()){
+            messages.add((String)entity.getProperty("comment"));
+        }
+
         String json = convertToJsonUsingGson(messages);
         response.setContentType("text/html;");
         response.getWriter().println(json);
     }
 
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        String comment = request.getParameter("comment");
+        long timestamp = System.currentTimeMillis();
+        
+        if(!comment.equals("")){
+            Entity commentEntity = new Entity("Comment");
+            commentEntity.setProperty("comment", comment);
+            commentEntity.setProperty("timestamp", timestamp);
```

Comment:
Optional: you might consider using Autovalue (https://github.com/google/auto/tree/master/value) to create a value class to manage interactions with comments, and conversion to/from entities. It's not required, but it would make things more readable and centralize the comment/entity management logic from the business logic of what you do with them.

Reply:
I will take a look at AutoValue, but in the essence of time, I won't be implementing it. Thank you for the suggestion though